src/document-converter.ts

```diff
@@ -274,13 +309,24 @@ export class DocumentConverter {
     removeUnnecessaryEventListeners(program);
     removeWrappingIIFEs(program);
     const importedReferences = this.collectNamespacedReferences(program);
+    const results: ConversionResult[] = [];
+
     // Add imports for every non-module <script> tag to just import the file
     // itself.
-    for (const scriptImport of this.document.getFeatures(
-             {kind: 'html-script'})) {
+    for (const scriptImport of this.document.getFeatures({ kind: 'html-script' })) {
       const oldScriptUrl = getDocumentUrl(scriptImport.document);
       const newScriptUrl = this.convertScriptUrl(oldScriptUrl);
-      importedReferences.set(newScriptUrl, new Set());
+      if (convertedHtmlScripts.has(scriptImport)) {
+        results.push({
```

Comment:
leave a comment that JS files with the same path as the convertedFilePath of _this_ file are deleted here, but re-written by the result pushed below, because the ordering of results matters.